src/components/cylc/tree/cylc-tree.js

```diff
@@ -14,14 +14,39 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-import { computePercentProgress } from '@/components/cylc'
 import { extractGroupState } from '@/utils/tasks'
 import { merge } from 'lodash'
 import { createFamilyProxyNode, getCyclePointId } from '@/components/cylc/tree/index'
 import TaskState from '@/model/TaskState.model'
 
 export const FAMILY_ROOT = 'root'
 
+/**
+ * Compute percent progress.
+ *
+ * @see https://github.com/cylc/cylc-flow/blob/de7d938496e82dbdfb165938145670dd8e801efd/lib/cylc/gui/updater_tree.py#L248-L263
+ * @param {number|undefined} startedTime in milliseconds since 1970-01-01 00:00:00 UTC, e.g. 1568353099874
+ * @param {number|undefined} meanElapsedTime mean elapsed time in seconds
+ * @returns {number} the percent progress, e.g. 25 (meaning 25% progress)
+ * @private
+ */
+function computePercentProgress (startedTime, meanElapsedTime) {
+  // startedTime > now reportedly possible via interaction with `cylc reset`
+  if (!startedTime || !meanElapsedTime) {
```

Comment:
Maybe a comment to explain why we need this? (mean elapsed time is undefined until the first instance of the task has finished running).

Reply:
Yup, good idea. Added a comment.

Reply:
I'd rather it gave the reason why undefined is possible, for future reference: task proxies exist before they start running, so startedTime will be undefined until then; and "mean elapsed time" is necessarily undefined until the first instance of a task has completed running (before that, nothing has elapsed to compute the mean of).

Reply:
I've edited my last commit @hjoliver . Hope you don't mind I've copied your text as-is :smile: Just added an extra line about the NaN. Also moved the `|| meanElapsedTime === 0` to this `if` statement. So the other `if` is only for the `startedTime > now`.